[Bug] Failed cleanup  mark test as passed

### chainsaw version Version

v0.2.13

### Description

With v0.2.13, when a step cleanup is ERROR, the report mark the test as "Passed" and the return code is 0 instead of "failed" (and return 1).

With v0.2.12, the report is "Failed" as expected.

### Steps to reproduce

I deploy an Ingress with a finalizer that chainsaw can't delete :
```yaml
- name: deploy-ingress
  try:
  - apply:
      file: samples/ingress.yaml
  - assert:
      file: samples/ingress.yaml
```
In the output I have that:
```
| 01:34:41 | my-test | deploy-ingress      | CLEANUP   | ERROR |
        === ERROR
        Get "https://10.308.194.15/apis/networking.k8s.io/v1/namespaces/test-namespace/ingresses/ingress-sample": context deadline exceeded
| 01:34:41 | my-test | deploy-ingress      | CLEANUP   | END   |
...
--- FAIL: chainsaw (5.21s)
    --- FAIL: chainsaw/my-test (8.29s)
FAIL
Tests Summary...
- Passed  tests 1
- Failed  tests 0
- Skipped tests 0
Done.
```

### Expected behavior

I did the exact same test with v0.2.12 and this is the output:
```
    | 01:31:24 | my-test | deploy-ingress      | CLEANUP   | ERROR |
        === ERROR
        Get "https://10.308.194.15/apis/networking.k8s.io/v1/namespaces/test-namespace/ingresses/ingress-sample": context deadline exceeded
    | 01:31:24 | my-test | deploy-ingress      | CLEANUP   | END   |
...
--- FAIL: chainsaw (5.18s)
    --- FAIL: chainsaw/my-test (8.82s)
FAIL
Tests Summary...
- Passed  tests 0
- Failed  tests 1
- Skipped tests 0
Done with failures.
Error: some tests failed
```

### Screenshots

_No response_

### Logs

```Shell

```

### Slack discussion

_No response_

### Troubleshooting

- [x] I have searched other issues in this repository and mine is not recorded.

--- a/pkg/runner/step_test.go
+++ b/pkg/runner/step_test.go
@@ -698,6 +698,13 @@ func TestStepProcessor_Run(t *testing.T) {
 							Duration: metav1.Duration{Duration: time.Duration(1) * time.Second},
 						},
 					},
+					{
+						PodLogs: &v1alpha1.PodLogs{
+							ActionObjectSelector: v1alpha1.ActionObjectSelector{
+								Selector: "name=myapp",
+							},
+						},
+					},
 				},
 				Finally: []v1alpha1.CatchFinally{
 					{
@@ -714,6 +721,13 @@ func TestStepProcessor_Run(t *testing.T) {
 					{
 						Sleep: &v1alpha1.Sleep{
 							Duration: metav1.Duration{Duration: time.Duration(1) * time.Second},
+						},
+					},
+					{
+						PodLogs: &v1alpha1.PodLogs{
+							ActionObjectSelector: v1alpha1.ActionObjectSelector{
+								Selector: "name=myapp",
+							},
 						},
 					},
 				},

